addons/obiaway/Obiaway.lua

```diff
@@ -137,9 +164,146 @@ function inTable(tbl, item)
     return false
 end
 
-function get_all_elements()
+-- converts name of a bag (str) to an id number. returns 0 (inventory) when no argument passed.
```

Comment:
Something like this can be done with this line:
```lua
local inv_ids = res.bags:rekey('english'):key_map(string.lower):map(table.get-{'id'})
```

Then you can use it like this:
```lua
print(inv_ids.storage)  -- Will output 2
```

One main advantages to this, a) it doesn't perform an expensive function call with many comparisons and b) if the IDs ever change it will automatically get them correct, since it reads them from the resources.